subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -1405,8 +1406,11 @@ static inline void ticker_job_worker_bh(struct ticker_instance *instance,
 #if !defined(CONFIG_BT_TICKER_COMPATIBILITY_MODE)
 		ticks_latency -= ticks_to_expire;
 
-		is_must_expire_skip = (ticker->must_expire &&
-				       (ticker->lazy_current != 0U));
+		/* Node with lazy count did not expire with callback, but
+		 * was either a collision or re-scheduled. This node should
+		 * not define the active slot reservation (slot_previous).
+		 */
+		skip_collision = ticker->lazy_current != 0U;
```

Comment:
With the addition of `ticker->ack--` in worker, the request state of a collided node is now 2, and since this is not a "must_expire" case, the _slot_previous will be updated unless guarded. It must be true (?) that if `lazy_current` is non-zero, the ticker did not expire in the "old" sense and went in the air, but rather just skipped further processing.